app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -163,11 +158,28 @@ define(function (require, exports, module) {
     },
 
     deleteDisplayedAccountProfileImage (account) {
-      return account.deleteAvatar(this._displayedProfileImage.get('id'))
-        .then(() => {
-          // A blank image will clear the cache
-          this.updateProfileImage(new ProfileImage(), account);
-        });
+      if ( ! account.get('profileImageId') ) {
+        return account.fetchCurrentProfileImage()
```

Comment:
@shane-tomlinson Hi, I have been studying about promises from past few days. @vladikoff gave me the intro link. 😄 
Also, is the code fine now?